elyra/pipeline/airflow/processor_airflow.py

```diff
@@ -442,19 +440,38 @@ def create_pipeline_file(self, pipeline, pipeline_export_format, pipeline_export
                 import black
                 autopep_output = autopep8.fix_code(python_output)
                 output_to_file = black.format_str(autopep_output, mode=black.FileMode())
+
                 fh.write(output_to_file)
 
         return pipeline_export_path
 
-    def _get_unique_operation_name(self, operation_name: str, operation_list: list) -> str:
-        unique_name_counter = 1
-        unique_operation_name = operation_name
-        names = [op['notebook'] for op in operation_list]
-        while unique_operation_name in names:
-            unique_name_counter += 1
-            unique_operation_name = ''.join([operation_name, '_', str(unique_name_counter)])
+    def _create_unique_node_names(self, operation_list, parent_unique_names=None):
```

Comment:
I think type hints would be useful here:
```suggestion
    def _create_unique_node_names(self, operation_list: List[Operation], parent_unique_names: Optional[Dict[str, int]] = None):
```

I'm not clear what should be done about the return value.  The way things stand is the return value is only used by the original caller and not recursive call - which is odd.